apps/native/app/src/stores/auth-store.ts

```diff
@@ -207,12 +220,18 @@ export async function checkIsAuthenticated() {
         intl.formatMessage({ id: 'login.expiredTitle' }),
         intl.formatMessage({ id: 'login.expiredScopesMessage' }),
       )
+
       await logout()
+
       return false
     }
   }
 
-  fetchUserInfo().catch(async (err) => {
+  try {
+    await fetchUserInfo()
+
+    return true
+  } catch (e) {
```

Comment:
This should mainly check if `fetchUserInfo()` fails due to `refresh()` failing with a 401 response. Anything else should be assumed as `authenticated === true`.

To test 401 refresh errors, try overriding the refreshToken with some garbage.

Reply:
Yes good point I added offline check as well